### PR TITLE
ci: Add mold to build tests

### PR DESCRIPTION
**Pull request by medhefgo**
This is on top of  #24253 to hopefully make ci happy.

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
           - { COMPILER: "gcc",   COMPILER_VERSION: "12", LINKER: "gold", CRYPTOLIB: "openssl" }
           - { COMPILER: "clang", COMPILER_VERSION: "12", LINKER: "gold", CRYPTOLIB: "gcrypt"  }
           - { COMPILER: "clang", COMPILER_VERSION: "13", LINKER: "lld",  CRYPTOLIB: "openssl" }
-          - { COMPILER: "clang", COMPILER_VERSION: "14", LINKER: "bfd",  CRYPTOLIB: "auto"    }
+          - { COMPILER: "clang", COMPILER_VERSION: "14", LINKER: "mold", CRYPTOLIB: "auto"    }
```
Comment (keszybz) [inline .github/workflows/build_test.yml]: Hmm, I think it'd be better to drop one of the `gold` tests instead. `gold` is on life-support, while `bfd` is very commonly used, so I think we should test both compilers with `bfd`.

**Comment by yuwata** [comment]
Maybe, this needs to be rebased?

**Comment by medhefgo** [comment]
I think the issue is that ubuntu mold is too old. It's failing at the lto build but it works locally on my arch with 1.4.0.

```
mold: error: src/udev/ata_id.p/ata_id_ata_id.c.o: looks like this is an LLVM bitcode, but mold does not support LTO
mold: error: LLVM LTO is detected, so falling back to ld.lld
mold: execvp failed: ld.lld: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by keszybz** [comment]
We get `mold amd64 1.0.3+dfsg-2`. Should this just wait until ubuntu is updated?

Also, maybe submit the second patch separately so we can merge it?

**Comment by medhefgo** [comment]
> We get `mold amd64 1.0.3+dfsg-2`. Should this just wait until ubuntu is updated?

Yeah, think so.

> Also, maybe submit the second patch separately so we can merge it?

The second patch doesn't make too much sense without the first…

**Review by bluca — APPROVED** [review]

**Pull request merged**